### DISCUSSION TITLE
[Question][Hw1] Boolean as i1: is boolean permitted for operands of arithmetic & comparison operators?

### DISCUSSION THREAD

**Discussion by retroinspect** [prosyslab-classroom/cs424-program-reasoning#394] [Q&A]
In the LLVM docs, i1 is a single-bit integer.
Since Bool type of interpreter is matched with i1 type, should we allow arithmetic & comparison operations for boolean operands?

**Comment by minseongg** [comment]
I think this is a related issue: https://github.com/prosyslab-classroom/cs524-program-analysis/issues/19

**Comment by retroinspect** [comment]
Thanks!
